packages/edit-site/src/hooks/use-theme-style-variations/use-theme-style-variations-by-property.js

```diff
@@ -71,27 +69,26 @@ export function useCurrentMergeThemeStyleVariationsWithUserConfig( {
 		};
 	}, [] );
 	const { user: baseVariation } = useContext( GlobalStylesContext );
+	const clondedBaseVariation = cloneDeep( baseVariation );
 
-	const variations = useMemo( () => {
+	const variationsWithSinglePropertyAndBase = variationsFromTheme
+		.filter( ( variation ) => {
+			return isVariationWithSingleProperty( variation, property );
+		} )
+		.map( ( variation ) => {
+			return mergeBaseAndUserConfigs( clondedBaseVariation, variation );
+		} );
```

Comment:
Move this inside the `useMemo`. As it is now, the `useMemo` is just pure overhead because this is creating a new array every time.